src/path_string.rs

```diff
@@ -20,30 +20,31 @@ impl PathString for serde_json::Value {
   }
 }
 
-// TODO: improve performance
-impl PathString for EvaluationContext<'_> {
-  fn path_string(&self, path: &[String]) -> Option<Cow<'_, str>> {
+fn convert_value(value: &async_graphql::Value) -> Option<Cow<'_, str>> {
+  match value {
+    async_graphql::Value::String(s) => Some(Cow::Borrowed(s.as_str())),
+    async_graphql::Value::Number(n) => Some(Cow::Owned(n.to_string())),
+    async_graphql::Value::Boolean(b) => Some(Cow::Owned(b.to_string())),
+    async_graphql::Value::Object(map) => Some(json!(map).to_string().into()),
+    async_graphql::Value::List(list) => Some(json!(list).to_string().into()),
+    _ => None,
+  }
+}
+
+impl<'a, Ctx: GraphqlContext<'a>> PathString for EvaluationContext<'a, Ctx> {
+  fn path_string<T: AsRef<str>>(&self, path: &[T]) -> Option<Cow<'_, str>> {
     let ctx = self;
-    let mut result = None;
-    if let Some((head, tail)) = path.split_first() {
-      result = match head.as_str() {
-        "value" => ctx.path_value(tail).map(|v| v.to_owned()),
-        "args" => ctx.args()?.get_path(tail).cloned(),
-        "headers" => ctx.get_header_as_value(&tail[0]),
-        "vars" => Some(async_graphql::Value::String(
-          ctx.req_ctx.server.vars.clone()?.get(&tail[0]).cloned()?,
-        )),
+
+    path.split_first().and_then(|(head, tail)| {
+      assert!(!tail.is_empty());
```

Comment:
Should return `none` instead of panicking.

Reply:
replaced here

But in the future, it might be worthwhile to add some validation of the correct definition of templates at the parsing stage.

Reply:
Agree 100%, would you mind creating a Github issue for this?

Reply:
created https://github.com/tailcallhq/tailcall/issues/449